mypy/test/teststubgen.py

```diff
@@ -103,26 +104,170 @@ def test_find_unique_signatures(self) -> None:
              ('func3', '(arg, arg2)')])
 
     def test_infer_sig_from_docstring(self) -> None:
-        assert_equal(infer_sig_from_docstring('\nfunc(x) - y', 'func'), ('(x)', 'Any'))
-        assert_equal(infer_sig_from_docstring('\nfunc(x, Y_a=None)', 'func'),
-                     ('(x, Y_a=None)', 'Any'))
-        assert_equal(infer_sig_from_docstring('\nafunc(x) - y', 'func'), None)
-        assert_equal(infer_sig_from_docstring('\nfunc(x, y', 'func'), None)
-        assert_equal(infer_sig_from_docstring('\nfunc(x=z(y))', 'func'), None)
-        assert_equal(infer_sig_from_docstring('\nfunc x', 'func'), None)
+        assert_equal(
+            infer_sig_from_docstring('\nfunc(x) - y', 'func'),
+            [TypedFunctionSig(
+                name='func',
+                args=[TypedArgSig(name='x', type=None, default=False)],
+                ret_type='Any'
+            )]
+        )
+
+        assert_equal(
+            infer_sig_from_docstring('\nfunc(x, Y_a=None)', 'func'),
+            [TypedFunctionSig(
+                name='func',
+                args=[
+                    TypedArgSig(name='x', type=None, default=False),
+                    TypedArgSig(name='Y_a', type=None, default=True)
+                ],
+                ret_type='Any'
+            )]
+        )
+        assert_equal(
+            infer_sig_from_docstring('\nfunc(x, Y_a=3)', 'func'),
+            [TypedFunctionSig(
+                name='func',
+                args=[
+                    TypedArgSig(name='x', type=None, default=False),
+                    TypedArgSig(name='Y_a', type=None, default=True)
+                ],
+                ret_type='Any'
+            )]
+        )
```

Comment:
As I said we use different style for multiline displays:
```python
        [TypedFunctionSig(name='func',
                          args=[TypedArgSig(name='x', type=None, default=False),
                                TypedArgSig(name='Y_a', type=None, default=True)],
                          ret_type='Any')]
```
this is much more compact.